restcomm/restcomm.dao/src/main/java/org/restcomm/connect/dao/entities/CallDetailRecordFilter.java

```diff
@@ -89,6 +91,25 @@ public CallDetailRecordFilter(String accountSid, List<String> accountSidSet, Str
         } else {
             this.instanceid = null;
         }
+
+        this.sortBy = mapAPIFieldsToDb(sortBy);
+        this.sortDirection = sortDirection;
+    }
+
+    // We want the fields that the API caller provides for sorting to have the exact same names as what
+    // is returned from the API when it returns CDRs in queries, so that the API is consistent and intuitive. However, when doing
+    // the actual query in the db we are using the DB column names which sometimes are different from those API-level
+    // fields. Let's use a mapping function to convert API-level fields to db column names.
+    public String mapAPIFieldsToDb(String sortBy) {
+        // TODO: One issue here is that whatever the API user sends inside SortBy is sent down to DB, which is messy. How is this currently handled in other cases in the API?
```

Comment:
i think any mapping to DB columns names should happen at DAO impl side, not in entities/RestEndpoint. So this is fair challenge

My suggestion is to add an specific sortByFrom sortByTo, and then check in DAO impl which ones are enabled, and set the query accordingly there.

Reply:
this is another bad example doing mapping in RESTEndpoint 
https://github.com/RestComm/Restcomm-Connect/blob/master/restcomm/restcomm.http/src/main/java/org/restcomm/connect/http/IncomingPhoneNumbersEndpoint.java#L341

Reply:
@jaimecasero, I lost you, right now the mapping is done in the Filter; not in the entity/endpoint, or am I getting the organization wrong?

Also, if I'm getting this right, you are talking about introducing in the filter API level multiple sortBy* attributes, one for each column we want to support sorting on, and potentially make them enums with possible values ASC or DESC? And the parsing from the initial URL string to those attributes would still go the Endpoint code, right?

Then, when you mention DAO impl for the mapping, are you talking about MybatisCallDetailRecordsDao.getCallDetailRecords()? But there isn't the filter already initialized and ready for use by the mybatis mapper .xml file? I'm getting the feeling that it will be weird to update the filter there

An example from the API codebase where you use what you describe would also be great.

If it's easier we can also have a call if you want to further discuss

Reply:
my initial point is that i dont want to find DB column names in REST or Entity/DAOInterface level. The Filter you mention is part of the Entity layer in this case, so to me seeing DB column names there is bad.

Column names should be the responsability of DAO impl to me. If we can agree on that, I may elaborate on my suggested API changes...

Reply:
This is already fixed per our discussion the other day, so we 're good